File: api/src/main/java/com/api/kookie/data/profile/ProfileRepository.java
package com.api.kookie.data.profile;

import com.api.kookie.data.entity.Profile;
import org.springframework.data.repository.CrudRepository;

public interface ProfileRepository extends CrudRepository<Profile, Integer> {

    Profile findOneByUser_Username(String username);

    Profile findOneByProfileId(Integer profileId);
}

File: mobile/kookie/lib/repositories/profile_repository.dart
import 'package:kookie/api/profile_api_client.dart';

class ProfileRepository {
  final ProfileApiClient profileApiClient = ProfileApiClient();

  Future<bool> addRecipeToFavorite(int profileId, int recipeId) async {
    return profileApiClient.addRecipeToFavorite(profileId, recipeId);
  }
}
